### PR TITLE
Update for merge of CrateDisambiguator into StableCrateId

### PR DESCRIPTION
**Pull request by bjorn3**
cc https://github.com/rust-lang/rust/pull/85804

### CODE REVIEW

File: src/backend/libs-and-metadata.md [hunk 1]
```diff
@@ -114,21 +114,22 @@ computed.
...
-By default, all Rust symbols are mangled and incorporate the disambiguator
-hash. This allows multiple versions of the same crate to be included together.
-Cargo automatically generates `-C metadata` hashes based on a variety of
-factors, like the package version, source, and the target kind (a lib and bin
-can have the same crate name, so they need to be disambiguated).
+By default, all Rust symbols are mangled and incorporate the stable crate id.
+This allows multiple versions of the same crate to be included together. Cargo
+automatically generates `-C metadata` hashes based on a variety of factors,
+like the package version, source, and the target kind (a lib and test can have
+the same crate name, so they need to be disambiguated).
```
Comment (bjorn3) [inline src/backend/libs-and-metadata.md:129]: If the bin crate type is used, this is included in the hash and as such a lib and bin never have the same hash. The comment about this describes this exact use case. This was true even before `CrateDisambiguator` was merged into `StableCrateId`.

**Comment by Mark-Simulacrum** [comment]
The relevant PR upstream might get reverted (https://github.com/rust-lang/rust/pull/85804#issuecomment-852158292) so we may want to hold off on this for a bit.

**Comment by JohnTitor** [comment]
FYI: revert revert PR (https://github.com/rust-lang/rust/pull/86143) is on review.

**Comment by JohnTitor** [comment]
The above PR has been merged 🎉 @bjorn3 IIUC this PR doesn't need an additional change and we can merge it now, right?

**Comment by bjorn3** [comment]
Correct.

**Review by jyn514 — APPROVED** [review]

**Pull request merged**